app/models/quota.rb

```diff
@@ -40,18 +40,26 @@ class Quota
   # @return [Time] when quota generated
   attr_reader :updated_at
 
+  # Number of files allowed over limit
+  # @return [Integer] file grace
+  attr_reader :file_grace
+
+  # Number of blocks allowed over limit
+  # @return [Integer] disk grace
+  attr_reader :disk_grace
+
   class << self
     # Get quota objects for all users in JSON file(s)
     def all(opts = {})
       find(nil, opts)
     end
 
     # Get quota objects only for requested user in JSON file(s)
-    def find(user, opts = {})
+    def find(quota_paths, user, opts = {})
```

Comment:
This method can be simplified further, by reducing its scope.

1. Instead of accepting multiple paths, just do 1 path
2. Instead of logging, raise an exception (custom named exception - here is an example of how to create custom exceptions https://github.com/OSC/ood_core/blob/396ec9e8720ac680fb11947e0e1e68b73c67ccd2/lib/ood_core/errors.rb)
3. No need for the opts hash, its not used anywhere so the method sig is now `find(quota_path, user)`

You need to change the one place this method is called, to call it on each path available and if an exception is raised, log it. And the result, two arrays, will need to be combined. But that doesn't take much code and can happen in one place outside of this class.